docs/samples/databricks/product_recommendation_demo.ipynb

```diff
@@ -0,0 +1,682 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Create training data using point-in-time correct feature join\n",
+        "\n",
+        "A training dataset usually contains entity id columns, multiple feature columns, event timestamp column and label/target column. \n",
+        "\n",
+        "To create a training dataset using Feathr, one needs to provide a feature join configuration file to specify\n",
+        "what features and how these features should be joined to the observation data. \n",
+        "\n",
+        "To learn more on this topic, please refer to [Point-in-time Correctness](https://github.com/linkedin/feathr/blob/main/docs/concepts/point-in-time-join.md)\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "if client.spark_runtime == 'databricks':\n",
+        "    output_path = 'dbfs:/feathrazure_test.avro'\n",
+        "else:\n",
+        "    output_path = feathr_output_path\n",
+        "\n",
+        "\n",
+        "feature_query = FeatureQuery(\n",
+        "    feature_list=[\"feature_user_age\", \n",
+        "                  \"feature_user_tax_rate\", \n",
+        "                  \"feature_user_gift_card_balance\", \n",
+        "                  \"feature_user_has_valid_credit_card\", \n",
+        "                  \"feature_user_total_purchase_in_90days\",\n",
+        "                  \"feature_user_purchasing_power\"], key=user_id)\n",
+        "settings = ObservationSettings(\n",
+        "    observation_path=\"wasbs://public@azurefeathrstorage.blob.core.windows.net/sample_data/product_recommendation_sample/user_observation_mock_data.csv\",\n",
+        "    event_timestamp_column=\"event_timestamp\",\n",
+        "    timestamp_format=\"yyyy-MM-dd\")\n",
+        "client.get_offline_features(observation_settings=settings,\n",
+        "                            feature_query=feature_query,\n",
+        "                            output_path=output_path)\n",
+        "client.wait_job_to_finish(timeout_sec=500)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Download the result and show the result\n",
+        "\n",
+        "Let's use the helper function `get_result_df` to download the result and view it:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "def get_result_df(client: FeathrClient) -> pd.DataFrame:\n",
+        "    \"\"\"Download the job result dataset from cloud as a Pandas dataframe.\"\"\"\n",
+        "    res_url = client.get_job_result_uri(block=True, timeout_sec=600)\n",
+        "    tmp_dir = tempfile.TemporaryDirectory()\n",
+        "    client.feathr_spark_launcher.download_result(result_path=res_url, local_folder=tmp_dir.name)\n",
```

Comment:
>feathr_spark_launcher [](http://example.com/codeflow?start=20&length=21)

I run this notebook and get error

2022-07-11 02:58:10.584 | DEBUG    | feathr._databricks_submission:wait_for_completion:180 - Current Spark job status: SUCCESS
AttributeError: 'FeathrClient' object has no attribute 'feathr_spark_launcher'